amaranth_boards/doppler.py

```diff
@@ -0,0 +1,71 @@
+from amaranth.build import *
+from amaranth.vendor.lattice_ice40 import *
+
+
+__all__ = ["DopplerPlatform"]
+
+
+# Dadamachines Doppler USB stick https://github.com/dadamachines/doppler/
+class DopplerPlatform(LatticeICE40Platform):
+    device      = "iCE40UP5K"
+    package     = "SG48"
+    default_clk = "SB_HFOSC"
+    hfosc_div   = 0
+    resources   = [
+        # PCB button1, button2
+        Resource("button", 0, Pins("2", dir="i", invert=True),
+                 Attrs(IO_STANDARD="SB_LVCMOS33", PULLUP=1)),
+        Resource("button", 1, Pins("3", dir="i", invert=True),
+                 Attrs(IO_STANDARD="SB_LVCMOS33", PULLUP=1)),
+
+        # LED matrix anodes and cathodes
+        Resource("aled", 0, Pins("26 27 28 31", dir="o"),
+                 Attrs(IO_STANDARD="SB_LVCMOS")),
+        Resource("kled", 0, Pins("42", dir="oe"), Attrs(IO_STANDARD="SB_LVCMOS")),
```

Comment:
It seems odd to request all anodes and individal cathodes as separate resources. I would make a resource led_matrix 0 and have subsignals `a` and `k`.